src/index.js

```diff
@@ -2,17 +2,23 @@ import ErrorBoundary from 'components/common/ErrorBoundary/ErrorBoundary';
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './pages/App';
-import { AuthProvider, SideDrawerProvider } from './providers';
+import { AuthProvider, NoteProvider, SideDrawerProvider, TagProvider } from './providers';
 import reportWebVitals from './reportWebVitals';
 import './styles/index.css';
 
 ReactDOM.render(
   <React.StrictMode>
     <ErrorBoundary>
       <AuthProvider>
-        <SideDrawerProvider>
-          <App />
-        </SideDrawerProvider>
+        {(user) => (
+          <NoteProvider user={user}>
+            <TagProvider user={user}>
+              <SideDrawerProvider>
+                <App />
+              </SideDrawerProvider>
+            </TagProvider>
+          </NoteProvider>
+        )}
       </AuthProvider>
     </ErrorBoundary>
```

Comment:
Instead of using all the providers on App.js itself, you can create <FlixNotesProvider> and add all these providers to it, and just use <FlixNotesProvider><App/></FlixNotesProvider> to make app.js leaner.